src/extract.ts

```diff
@@ -145,83 +138,71 @@ function getHumanReadableUnitValue(seconds: number): [number, string] {
-        url,
-    };
-}
+    const pr = github.context.payload.pull_request;
+
+    if (pr) {
+        // On pull_request hook, head_commit is not available
+        const id: string = pr.head.sha;
+        const username: string = pr.head.user.login;
+        const user = {
+            name: username, // XXX: Fallback, not correct
+            username,
+        };
+
+        return {
+            author: user,
+            committer: user,
+            id,
+            message: pr.title,
+            timestamp: pr.head.repo.updated_at,
+            url: `${pr.html_url}/commits/${id}`,
+        };
+    }
+
+    if (!githubToken) {
+        throw new Error(
+            `No commit information is found in payload: ${JSON.stringify(github.context.payload, null, 2)}. Also, no 'github-token' provided, could not fallback to GitHub API Request.`,
+        );
+    }
 
-async function getHeadCommit(githubToken: string): Promise<Commit> {
-    // On cron and manually dispatched workflows head_commit and pull_request is not available
-    // We try to get the data of current HEAD via GitHub REST
     const octocat = new github.GitHub(githubToken);
+
     const { status, data } = await octocat.repos.getCommit({
         owner: github.context.repo.owner,
         repo: github.context.repo.repo,
         ref: github.context.ref,
     });
-    if (status !== 200 && status !== 304) {
+
+    if (!(status === 200 || status === 304)) {
         throw new Error(`Could not fetch the head commit. Received code: ${status}`);
     }
 
     const { commit } = data;
 
-    const author = {
-        name: commit.author.name,
-        username: commit.author.name, // XXX: Fallback, not correct
-    };
-    const committer = {
-        name: commit.committer.name,
-        username: commit.committer.name, // XXX: Fallback, not correct
-    };
-
     return {
-        author,
-        committer,
+        author: {
+            name: commit.author.name,
+            username: data.author.login,
+            email: commit.author.email,
+        },
+        committer: {
+            name: commit.committer.name,
+            username: data.committer.login,
+            email: commit.committer.email,
+        },
         id: data.sha,
         message: commit.message,
         timestamp: commit.author.date,
         url: data.html_url,
     };
 }
 
-async function getCommit(githubToken?: string): Promise<Commit> {
-    if (github.context.payload.head_commit) {
-        return github.context.payload.head_commit;
-    }
-
-    if (github.context.payload.pull_request) {
-        return getCommitFromPr(github.context.payload.pull_request);
-    }
-
-    if (!githubToken) {
-        throw new Error(
-            `No commit information is found in payload: ${JSON.stringify(
-                github.context.payload,
-                null,
-                2,
-            )} and 'github-token' input is not set`,
-        );
-    }
-    return await getHeadCommit(githubToken);
-}
-
```

Comment:
Can you still leave this function in the split form with separate function calls to `getCommitFromPr` and `getHeadCommit`? I think splitting your logic into `getCommitFromPr` and `getHeadCommit` makes the code more readable and understandable.

Reply:
Sorry, I wouldn't agree that adding the extra functions makes it more readable. This split adds 2 additional functions which are comparatively shallow and never reused anywhere else in the code. A "deeper" function for `getCommit` means users of this interface only need to know this 1 function, knowing that this whole function works towards the goal of "getting the commit", without jumping to other functions which are never re-used (assume that the point of functions are to "re-use" code and not to create logical separations for code).

Nonetheless, I made the changes you requested, but would echo that I think it was most readable and sustainable the way it originally.